### PR TITLE
feat: dumbo body limit

### PR DESCRIPTION
**Pull request by darkterminal**
Related issue: #41

### CODE REVIEW

File: src/Helpers/BodyLimit.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Dumbo\Helpers;
+
+use Dumbo\Context;
+
+class BodyLimit
+{
+    public static function limit(int $maxSize, callable $onError = null): callable
```
Comment (notrab) [inline src/Helpers/BodyLimit.php]: Missing the doc block. Something like this?

```php
/**
     * Create a middleware that limits the size of the request body
     *
     * @param int $maxSize The maximum allowed size of the request body in bytes
     * @param callable|null $onError Custom error handler (optional)
     * @return callable The middleware
     */
```

File: src/Router.php [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 use FastRoute\Dispatcher;
 use FastRoute\RouteCollector;
+use Psr\Http\Message\ResponseInterface;
```
Comment (notrab) [inline src/Router.php:7]: Is this needed?

File: src/Helpers/BodyLimit.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        return function (Context $context, callable $next) use ($maxSize, $onError) {
+            $contentLength = $context->req->header('Content-Length');
+
+            if ($contentLength > $maxSize) {
```
Comment (notrab) [inline src/Helpers/BodyLimit.php]: @darkterminal what do you think about making the error handling more explicit with a method on the class?

We could change this to:

```php
if ($contentLength !== null && (int)$contentLength > $maxSize) {
  return self::handleError($context, $maxSize, $onError);
}
```

Then add the following:

```php
private static function handleError(Context $context, int $maxSize, callable $onError = null): ResponseInterface
{
    if ($onError !== null) {
        return $onError($context, $maxSize);
    }

    throw new HTTPException(
        413,
        "Request entity too large. The body must not be larger than $maxSize bytes.",
        "BODY_TOO_LARGE",
        ["max_size" => $maxSize]
    );
}
```

**Review by notrab — COMMENTED** [review]
Looks great @darkterminal! I made a few comments 👯

File: src/Helpers/BodyLimit.php (hunk 1 above)
Comment (darkterminal) [reply src/Helpers/BodyLimit.php]: Ah yes!

File: src/Router.php (hunk 2 above)
Comment (darkterminal) [reply src/Router.php:7]: Why this is everywhere... I am already answer this :rofl:

File: src/Helpers/BodyLimit.php (hunk 3 above)
Comment (darkterminal) [reply src/Helpers/BodyLimit.php]: It's should!

File: src/Router.php (hunk 2 above)
Comment (notrab) [reply src/Router.php:7]: @darkterminal I'm sorry, but I reviewed everything all at once.

**Comment by darkterminal** [comment]
The song entitled Body Limit has finished playing... :smile:

**Review by notrab — APPROVED** [review]
Looks great @darkterminal! I just updated the example to follow the other ones (default response + custom).

**Pull request merged**